### ISSUE TITLE
Updating from 0.3.3 to 0.4.1, it breaks my charts (LineChart and PieChart)

### ISSUE THREAD

**Issue by jamesblasco** [imaNNeoFighT/fl_chart#100]
Given a size to the father container of my chart it throws this error.

The following assertion was thrown building PieChart(duration: 150ms, dirty, dependencies: [MediaQuery], state: PieChartState#f59ef(ticker active)):
RenderBox was not laid out: RenderCustomPaint#99bfc NEEDS-LAYOUT NEEDS-PAINT 'package:flutter/src/rendering/box.dart': Failed assertion: line 1687 pos 12: 'hasSize'

**Comment by imaNNeoFighT** [comment]
I will check,
Thanks

**Comment by jamesblasco** [comment]
I will try to research more about it and give a proper example. It happens when i resize de view for example when i rotate the device

**Comment by imaNNeoFighT** [comment]
Ok prepare a reproducers sample, It helps me to debug it faster.

**Comment by jamesblasco** [comment]
I found when does it happen but don't know how to solver it. 

It happens when the chart is inside a LayoutBuilder widget. 
I replaced the _MyHomePageState of the fl_chart example with the following code and it throws the error.


```dart
class _MyHomePageState extends State<MyHomePage> {
 
  @override
  Widget build(BuildContext context) {
    return LayoutBuilder(
        builder: (BuildContext context, constraints) => Center(
            child: FractionallySizedBox(
                heightFactor: 0.5,
                widthFactor: 0.5,
                child: Scaffold(
                  body: SafeArea(
                    child: PageView(
                      children: <Widget>[
                        LineChartPage(),
                        BarChartPage(),
                        BarChartPage2(),
                        PieChartPage(),
                        LineChartPage2(),
                        LineChartPage3(),
                        LineChartPage4(),
                      ],
                    ),
                  ),
                ))));
  }
}
```

**Comment by imaNNeoFighT** [comment]
Dude you are trying to show the whole page with a 0.5 scale factor.
Is it intentionally?

**Comment by jamesblasco** [comment]
Yes, sorry i was testing it on ipad and i wanted to check the resize when rotating. But I think for the example is not needed

**Comment by jamesblasco** [comment]
The error happens in [containerRenderBox.size ](https://github.com/imaNNeoFighT/fl_chart/blob/9cf12867448cd9907dc65b2aff12dbc95be8a9eb/lib/src/chart/line_chart/line_chart.dart#L132) when the chart is a child of an upper tree that contains a LayoutBuilder.

**Comment by jamesblasco** [comment]
My usecase of LayoutBuilder: 
In my project I use LayoutBuilder to display different layout for this cases portrait or landscape and smartphone or ipad.

**Comment by imaNNeoFighT** [comment]
Run it with scale factor 1.0,
It works perfectly

**Comment by jamesblasco** [comment]
Not to me
![ezgif-6-156c10e6c7c0](https://user-images.githubusercontent.com/19904063/68548559-6df39080-03ee-11ea-8449-1a04170ca82b.gif)

**Comment by jamesblasco** [comment]
I think it is related to this :

[widgets > BuildContext > findRenderObject ](https://api.flutter.dev/flutter/widgets/BuildContext/findRenderObject.html)
> This method will only return a valid result after the build phase is complete. It is therefore not valid to call this from a build method. It should only be called from interaction event handlers (e.g. gesture callbacks) or layout or paint callbacks.

>If the render object is a RenderBox, which is the common case, then the size of the render object can be obtained from the size getter. This is only valid after the layout phase, and should therefore only be examined from paint callbacks or interaction event handlers (e.g. gesture callbacks).

In this part of the code the RenderBox comes from the findRenderObject(). So probably when rotating it tries to get the size of the render box while the layout is being built.
Size _getChartSize() {
    if (_chartKey.currentContext != null) {
      final RenderBox containerRenderBox = _chartKey.currentContext.findRenderObject();
      return containerRenderBox.size;
    } else {
      return getDefaultSize(context);
    }
  }

**Comment by jamesblasco** [comment]
I replaced `containerRenderBox.size` to `containerRenderBox.constraints.biggest` and it works.

I don't understand much de `getDefaultSize` and why it is a square 70% of the screen

**Comment by imaNNeoFighT** [comment]
Good catch!
I appreciate it, and please make a PR and help to fix it, it's valuable for the community.
`getDefaultSize()` is just return a default size (when there is no constraint for the chart, for example when you try to show it directly as a centered widget).
but the best case is to size it yourself.
(for example, you can show a `Container` that fills the screen (default size), but sometimes you need to set the size manually).

**Comment by ica4c** [comment]
Good evening. I've tried @jamesblasco solution over a LayoutBuilder and it didn't work. Got same NEEDS-LAYOUT NEEDS-PAINT debug message. What I found though when I comment out the chartKey at CustomPaint of e.g LineChart class everything start to work as expected I haven't spotted any differences in behavior too. What it actually do? Is it crucial in this context? Might be a dumb question, sorry in advance)
